logs/logs.go

```diff
@@ -101,14 +103,26 @@ func (l *LogAgent) Run(ctx context.Context) {
 				srcs := c.FindLogSrc()
 				for _, src := range srcs {
 					dname := src.Destination()
+					logGroup := src.Group()
+					logStream := src.Stream()
+					description := src.Description()
+					retention := src.Retention()
 					backend, ok := l.backends[dname]
 					if !ok {
-						log.Printf("E! [logagent] Failed to find destination %v for log source %v/%v(%v) ", dname, src.Group(), src.Stream(), src.Description())
+						log.Printf("E! [logagent] Failed to find destination %s for log source %s/%s(%s) ", dname, logGroup, logStream, description)
 						continue
 					}
-					dest := backend.CreateDest(src.Group(), src.Stream(), src.Retention())
+
+					if retention > 0 && l.retentionAlreadySet[logGroup] {
+						log.Printf("D! [logagent] Retention already set for log group %s, current retention %d", logGroup, retention)
+						retention = -1
+					} else if retention > 0 {
+						log.Printf("I! First time setting retention for log group %s, update map to avoid setting twice", logGroup)
+						l.retentionAlreadySet[logGroup] = true
+					}
```

Comment:
I know it's late to ask for this, but I'd feel better if we added a unit test specifically for this? Not as part of the translation, but specifically this part where we are modifying a map in memory. I would pull out this chunk of code into a separate function. 

It's pretty straightforward so I can live without it, but on the flip side, it wouldn't be a lot of extra work to include tests around it.